include/ccf/base_endpoint_registry.h

```diff
@@ -83,8 +83,8 @@ namespace ccf
 
     /** Get the status of a transaction by ID, provided as a view+seqno pair.
      * This is a node-local property - while it will converge on all nodes in
-     * a healthy network, it is derived from distributed state rather than
-     * distributed itself.
+     * a healthy network, it is either derived from distributed state in BFT,
+     * or replicated out by the primary in CFT.
```

Comment:
GitHub just lost a longer comment here.

I agree the original comment is a little unclear, but I think this is missing an important distinction between replicated (from the KV) and node-local (everything else) state.

"What is the value of key X in table Foo at TxID M.N?" => A question about replicated state, will get the same answer on ever node (if they know M.N)
"Is M.N committed?" => A derived node-local property, which may differ across nodes (even nodes which all _have_ M.N, and could answer the first question identically)

We have stronger consistency + serializability claims about the state that is directly replicated (we'll tell you a fact that exists at a specific TxID) than anything that is node-local. I know the main source of information for this bit of node-local state _is_ replicated by the primary, but I think the local thing is distinct, and derived (it's not the primary's last commit index, it's that combined with the last index we have locally, which means we may report a global commit index different from anything the primary ever claimed, because it advanced in different batches).

Reply:
I guess what I want here is a strong warning that this API, and anything else that accesses consensus/node-local things, is _not_ transactional and may not do what the caller expects. But the replicated/derived language is a step removed from that. What we really want is a disclaimer that doing:

```
status1 = get_status_v1(ARGS);
time.sleep(5);
status2 = get_status_v1(ARGS);
assert(status1 == status2);
```

might fail.

Reply:
Is the new phrasing better?

Reply:
Yes, I think the new phrasing covers all the important details!